packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1687,6 +1704,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 summaryLogger,
             });
 
+            const summary = summarizeResult.summary.tree;
+            const summaryTree = summary[channelsTreeName] as ISummaryTree;
+
+            const count = this.countDataStore(summaryTree.tree);
+            summaryLogger.sendTelemetryEvent({eventName: "dataStoreCount"}, count);
```

Comment:
Instead of adding new event, can you add it to a set of properties returned from this function (GenerateSummaryData type)? They will be automatically included in telemetry. See code in generateSummaryWithLogging() building telemetryProps.